### PR TITLE
Rename polygon weth and eth pos

### LINKED ISSUE #2583: Polygon in-wallet swaps to WETH
### Discord Discussion Link

https://discord.com/channels/808358975287722045/1037670242421321769/1040170198088556575

### What browsers are you seeing the problem on?

Chrome

### What were you trying to do?

In-wallet swap to WETH

### What did not work?

**When swapping to WETH on Polygon, an in-wallet swap swaps to**
- Wrapped Ethereum (PoS) https://polygonscan.com/token/0xae740d42e4ff0c5086b2b5b5d149eb2f9e1a754f
- Example in-wallet swap tx: https://polygonscan.com/tx/0xd99f973810754dbfb125016ebcd1ea0134acbf4f4fbbc864740fcefcf719fb06

**A swap on Matcha to WETH, swaps to**
- Wrapped Ether (Wormhole) :https://polygonscan.com/token/0x11cd37bb86f65419713f30673a480ea33c826872
- Example tx: https://polygonscan.com/tx/0xd54f2e851cee8c8533a5f0c4ab8fd7cb37aee6d76a6a0800ce76ba8dd716f8f1

**A swap on Uniswap to WETH** swaps to
- Wrapped Ether: https://polygonscan.com/token/0x7ceb23fd6bc0add59e62ac25578270cff1b9f619
- Example tx: https://polygonscan.com/tx/0x1105337e2a4394a0771289ab96011c870f472c3fd59307524884bb030898f984
### Version

v0.17.7

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by greg-nagy**
# What? 

Rename the current `ETH - POS` and `Weth` tokens from the polygon token list to match the user's expectations. 

# Test

- Test the fix on new install / fresh fetch
  - install wallet 
  - open devtools > Application > indexedDB > `tally/indexing` > tokenLists
  - check that in all the instances where the name is `Popular Tokens` list > 
    - ⚠️ There should be 2 of these in the db. The 2nd one arrives a bit after the install. To see it, you need to refresh the db view! CleanShot 2022-11-10 at 11 54 13
    - tokens[0] CleanShot 2022-11-10 at 11 51 46
    - tokens[28] CleanShot 2022-11-10 at 11 52 09
- Test the migration
  - remove the extension 
  - comment out [these lines](https://github.com/tallycash/extension/blob/83cb6ad88d3b19797ccdeb822a33ade2b37ba98e/background/services/indexing/index.ts#L712-L716)
  - `yarn start` > install the extension
  - check the same thing as in the previous point 
- Check UI 
  -  on a fresh install 
  - import seed 
  - on swap page
  - check you have 
    - CleanShot 2022-11-10 at 11 58 01
    - CleanShot 2022-11-10 at 11 58 20
    - swap to `WETH` and check it on polygonscan
      - [example tx](https://polygonscan.com/tx/0x9db07e80182b737fb48c4c87c145b006c9c07eb37216771a0a19b1e1604174cf)

Closes #2583

Latest build: [extension-builds-2586](https://github.com/tallycash/extension/suites/9233860812/artifacts/431598179) (as of Thu, 10 Nov 2022 12:07:52 GMT).

### CODE REVIEW

File: background/services/indexing/token-list-edit.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import { TokenInfo } from "@uniswap/token-lists"
+
+export const polygonTokenListURL =
+  "https://api-polygon-tokens.polygon.technology/tokenlists/default.tokenlist.json"
+
+export function fixPolygonWETHIssue(tokensParam: TokenInfo[]): TokenInfo[] {
+  // See the discussion behind this migration here
+  // https://matrix.to/#/!WZRftnsYNnOtzOGuZy:thesis.co/$nhJA20S7sFCv8qyPoxsYF0xIlOBAW4rqhdkUEmxCQYQ?via=thesis.co&via=matrix.org
```
Comment (jagodarybacka) [inline background/services/indexing/token-list-edit.ts:8]: element link is not very reliable, I would prefer short explanation what happened

Comment (greg-nagy) [reply background/services/indexing/token-list-edit.ts:8]: yes and no :)

**Review by jagodarybacka — APPROVED** [review]
Looking good ✅

**Review by jagodarybacka — APPROVED** [review]
✅

**Pull request merged**